gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientSettings.java

```diff
@@ -208,6 +219,7 @@ public B setBackgroundExecutorProvider(ExecutorProvider executorProvider) {
 
     /** Sets the CredentialsProvider to use for getting the credentials to make calls with. */
     public B setCredentialsProvider(CredentialsProvider credentialsProvider) {
+      explicitCredentialsProvided = true;
```

Comment:
unfortunately no because that could have been set as the default CredentialProvider.  [The requirements state in that scenario](https://docs.google.com/document/d/13qovLzgsk_Rb5nizYnjDDvJ6jL_mIG7MUIadvlV7SqM/edit?resourcekey=0-rg7PSjFwP5PrFpJLsmBS8A&tab=t.0#bookmark=id.h83h02o5ksek) we should just ignore the default credential provider.  I can leave a comment in the code, but lets also discuss in weekly meeting to ensure we're on same page

Reply:
a couple of options:

1) Most thorough: we add tracking to generated ServiceStubSettings classes to include tracking the defaultCredentialProvider object and check to see if current credentialProvider object matches and if so ignore.  Only negative is this will add a bit of generated code which is only necessary for this validation.

2) Best compromise: track if ClientSettings.setCredential was called and if not check if credentialProvider is NOT GoogleCredentialsProvider.  If either of these is true and api key is set, throw error.
While this will not catch a user using StubSettings to set the credential provider, that is not the published [java docs samples of usage](https://github.com/googleapis/google-cloud-java/blob/main/java-language/google-cloud-language/src/main/java/com/google/cloud/language/v2/LanguageServiceClient.java#L186)

3) Simplest: treat any GoogleCredentialsProvider as default credentials.

Reply:
removed validation for now.